### PR TITLE
Joshfischer/eco docs

### PR DESCRIPTION
**Pull request by joshfischer1108**
This pull request is not ready for review.  I am only opening now in case of needing feedback based on my current code.  This pull request contains:

- [ ] Documentation on how use ECO to be viewed on the static site
- [ ] The ability to add component level ram and jvm configs
- [ ] Property Substitution, either by environment variables or properties files passed in via command line.

I will check tasks off as I have them completed.

### CODE REVIEW

**Comment by joshfischer1108** [comment]
@lucperkins or @kramasamy Can either of you look at this error that occurred during the build?  I think I've seen it before when building out the docs.  `error: Server does not allow request for unadvertised object 87d417e53308ef794b53734bc09f86f01d1a0ace`.  Everything built fine on my local.  No rush, when ever you have time.

File: website/data/alerts.yaml [hunk 1]
```diff
@@ -23,4 +23,8 @@ new-streamlet-api:
   content: |
     As of version 0.16.0, Heron offers a new **Streamlet API** that you can use to write topologies in a more declarative, functional manner, without needing to specify spout and bolt logic directly. The Streamlet API is **currently in beta** and available for [Java](../../developers/java/streamlet-api). The Streamlet API for Python will be available soon.
 
-    More information on the Streamlet API can be found [below](#the-heron-streamlet-api).
\ No newline at end of file
+    More information on the Streamlet API can be found [below](#the-heron-streamlet-api).
+eco-api-beta:
+  title: The Heron ECO API is in beta
+  content: The Heron ECO API can be used to build and test topologies locally.  The API still needs some testing and feedback from the community to understand how we  should continue to develop ECO.
```
Comment (kramasamy) [inline website/data/alerts.yaml]: @joshfischer1108 - API can be used for topologies running as well in cluster - correct?

File: website/scripts/python-doc-gen.sh [hunk 2]
```diff
@@ -5,8 +5,8 @@ HERON_ROOT_DIR=$(git rev-parse --show-toplevel)
 INPUT=heronpy
 TMP_DIR=$(mktemp -d)
 
-sudo pip install heronpy==${HERONPY_VERSION}
-sudo pip install --ignore-installed six
+sudo -H pip install heronpy==0.17.1
```
Comment (kramasamy) [inline website/scripts/python-doc-gen.sh]: @lucperkins - is this change ok?

Comment (joshfischer1108) [reply website/scripts/python-doc-gen.sh]: I will back this change out.  I had to do some tricky stuff to get the pages to build on my local.  I meant to revert this, it slipped my mind.  Good catch.  Thank you

File: website/data/alerts.yaml (hunk 1 above)
Comment (joshfischer1108) [reply website/data/alerts.yaml]: @kramasamy Yes, just a typo error. I will fix

**Comment by joshfischer1108** [comment]
@kramasamy All requested have been made.

**Comment by lucperkins** [comment]
@joshfischer1108 Okay, I see the issue in CI. Run the following and then commit and push the resulting changes:

```bash
$ rm -rf website/public
$ git submodule update --init
```

In general, don't commit changes to that `website/public` folder, as the version control and publishing processes are separate.

**Comment by joshfischer1108** [comment]
@lucperkins I cannot get these pages to build locally on my machine, that is why I had to push them to the `gh-pages` branch.  I spent quite a bit of time trying to get this to work.   I will try again.  If not I will mention you and ask for assistance. Ok with you?

**Comment by lucperkins** [comment]
@joshfischer1108 Please feel to get in touch (luc@streaml.io) about issues you've had getting set up, For now, though, this PR will not pass CI if there are changes to that folder.

**Pull request closed without merging**